usercommandmgmt/usercommandmgmt.py

```diff
@@ -186,25 +193,138 @@ def get_highest_user_comm_allowance(self, member):
 
         return max(rel_usr_roles)
 
-    @staticmethod
-    def mod_evaluate_command(self, proposed_msg):
+    async def submit_for_approval(self, ctx, command, text):
         """
         posts the proposed command in the specified channel, and waits for a moderator's reaction to either
-        reject or accept the command. Will  return true if command_moderation is set to False or command
-        passes moderator evalutaion. False if it is rejected by a moderator.
-        Note: proposed_msg is the proposed command
+        reject or accept the command.
         """
-        if not command_moderation:
-            return True
-        else:
-            # submits command for evaluation and awaits response; TODO
-            pass
+        # create message to submit for mod approval
+        message_to_submit = (
+            "Sender: "
+            + ctx.author.name
+            + "\n"
+            + "Proposed Command: "
+            + "\n"
+            + command
+            + "\n"
+            + text
+            + "\n \n"
+            + ctx.message.jump_url
+        )
+
+        # finds the channel to send this message to: the moderator one; specified in configurable.py
+        mod_channel = self.find_channel_by_name(
+            mod_channel_name, ctx.message.guild.channels
+        )
+        # submit proposed command and relavant info to specified moderation channel
+        msg = await mod_channel.send(message_to_submit)
+        # adds reacts
+        await msg.add_reaction("✅")
+        await msg.add_reaction("❌")
+        self.create_comamnd_queue.append([msg, ctx, command, text])
+
+    def find_channel_by_name(self, channel_name, channel_List):
 
-    @staticmethod
-    def enforce_user_cmd_limit(self, member, server):
         """
-        returns true if the current number commands owned by the user is less than the highest amount allowed by any of their roles.
+        finds the discord channel in the list with the given name, and returns it. Will return none if a channel can't be found
+        """
+
+        for i in channel_List:
+            if i.name == channel_name:
+                return i
+
+        return None
+
+    def find_command_req_data(self, msg):
+        """
+        finds the given command request data from the queue by matching its message with the given one
         """
-        return self.activeDb.GetUserCommQuantity(
-            member.id, server_id=server.id
-        ) < get_highest_user_comm_allowance(self, member=member)
+        for req in self.create_comamnd_queue:
```

Comment:
Probably won't have any practical effects, but do consider if you used a dict instead of a list for the command queue. Might have a better runtime? Again, probably no practical effect but I think it's a good exercise to think about the implications (pros/cons) of using a different data structure here.

This part is fine as it is though.